theonion/bulbs-public-ads-manager
# Fix (ads) - Return the type of Social Referrer for Splashy Top

**What is it?**
This PR is to allow the type of social media referrer site to be properly listed for SPLASHY_TOP and bulbs ads going forward. Right now, if use comes from a social media site, we just list `socialReferrer=true`. With this change, if they come from one of these six sites (facebook, twitter, tumblr, reddit, pinterest, and instagram), we'll list it like `socialReferrer=facebook`.

**How to test locally**:
- Locally run `mantle` and in your code editor, find the `node_modules` folder
- Within `node_modules`, find `bulbs-public-ads-manager/src/helpers/SocialReferrer.js`
- Copy the changes in this PR into `SocialReferrer.js`, and save the file
- Open browser and navigate to https://facebook.com/ or https://pinterest.com/
- Open Dev Tools Console
- Run this command to visit a page:
`window.location.href='http://deadspin.localhost:9000/my-favorite-subplot-of-winnipeg-vegas-is-people-trying-1826037876?adzone=kinjatest';`
- In Dev Tools, switch to `Network` tab.
- Filter in the `Network` tab for "gamp"
- Select the third call, then check the "Query String Parameters" for `prev_scp`
- Within `prev_scp` you should see `socialReferrer=facebook`

<img width="748" alt="screen shot 2018-05-16 at 4 04 59 pm" src="https://user-images.githubusercontent.com/36833757/40144145-39870a42-5923-11e8-9139-9482b3a9e520.png">


**Asana Ticket:**
https://app.asana.com/0/366985625403531/663032124977691
